client-v2/src/main/java/com/clickhouse/client/api/Client.java

```diff
@@ -1023,7 +1043,41 @@ private void setDefaults() {
             if (!configuration.containsKey(ClientSettings.HTTP_USE_BASIC_AUTH)) {
                 useHTTPBasicAuth(true);
             }
+
+            String userAgent = configuration.getOrDefault(ClientSettings.HTTP_HEADER_PREFIX + HttpHeaders.USER_AGENT.toUpperCase(Locale.US), "");
+            String clientName = configuration.getOrDefault(ClientSettings.CLIENT_NAME, "");
+            httpHeader(HttpHeaders.USER_AGENT, buildUserAgent(userAgent.isEmpty() ? clientName : userAgent));
+        }
+
+        private static String buildUserAgent(String customUserAgent) {
+
+            StringBuilder userAgent = new StringBuilder();
+            if (customUserAgent != null && !customUserAgent.isEmpty()) {
+                userAgent.append(customUserAgent).append(" ");
+            }
+
+            userAgent.append(CLIENT_USER_AGENT);
+
+            String clientVersion = Client.class.getPackage().getImplementationVersion();
+            if (clientVersion == null) {
+                clientVersion = LATEST_ARTIFACT_VERSION;
+            }
+            userAgent.append(clientVersion);
+
+            userAgent.append(" (");
+            userAgent.append(System.getProperty("os.name"));
+            userAgent.append("; ");
+            userAgent.append("jvm:").append(System.getProperty("java.version"));
+            userAgent.append("; ");
+
+            userAgent.setLength(userAgent.length() - 2);
+            userAgent.append(')');
+
+            return userAgent.toString();
         }
+
+        public static final String LATEST_ARTIFACT_VERSION = "0.7.1-patch1";
+        public static final String CLIENT_USER_AGENT = "ch-j-v2/";
```

Comment:
I'd prefer the full form `clickhouse-java-v2`

Reply:
changed